monai/transforms/utility/array.py

```diff
@@ -166,6 +168,58 @@ def __call__(self, img):
         return np.squeeze(img, self.dim)
 
 
+class DataStats(Transform):
+    """
+    Utility transform to show the statistics of data for debug or analysis.
+    It can be inserted into any place of a transform chain and check results of previous transforms.
+    """
+    def __init__(
+        self,
+        prefix='Data',
+        data_shape=True,
+        intensity_range=True,
+        data_value=False,
+        additional_info: Callable = None,
+        logger_handler=None
+    ):
+        """
+        Args:
+            prefix (string): will be printed in format: "{prefix} statistics".
+            data_shape (bool): whether to show the shape of input data.
+            intensity_range (bool): whether to show the intensity value range of input data.
+            data_value (bool): whether to show the raw value of input data.
+                a typical example is to print some properties of Nifti image: affine, pixdim, etc.
+            additional_info (Callable): user can define callable function to extract additional info from input data.
+            logger_handler (logging.handler): add additional handler to output data: save to file, etc.
+                add existing python logging handlers: https://docs.python.org/3/library/logging.handlers.html
+        """
+        assert isinstance(prefix, str), 'prefix must be a string.'
+        self.prefix = prefix
+        self.data_shape = data_shape
+        self.intensity_range = intensity_range
+        self.data_value = data_value
+        if additional_info is not None:
+            assert isinstance(additional_info, Callable), 'additional_info must be a Callable function.'
+        self.additional_info = additional_info
+        self.output = None
+        logging.basicConfig(level=logging.NOTSET)
+        self._logger = logging.getLogger('DataStats')
+        if logger_handler is not None:
+            self._logger.addHandler(logger_handler)
+
+    def __call__(self, img):
```

Comment:
For this method I would suggest not using inline conditional expressions as they get a little hard to read. This would be a good place to use the f-strings as well, eg.:

```python
def __call__(self, x):
    lines = ['']
    
    if self.data_shape:
        lines.append(f'Shape: {x.shape}')
        
    if self.intensity_range:
        lines.append(f'Intensity range: ({np.min(x)}, {np.max(x)})')
        
    if self.data_value:
        lines.append(f'Value: {x}')
    
    if self.additional_info:
        lines.append(f'Additional info: {self.additional_info(x)}')
        
    self.output = f'{self.prefix} statistics:{"\n  ".join(lines)}'
    self._logger.debug(self.output)
    
    return x
```

This is a style choice perhaps, the rest of the PR looks good.

Reply:
Also, when `self.data_value` is True, without a sanity check to prevent printing a value that's way too large we're leaving it up to the user to prevent this.

Reply:
Hi @ericspod ,

Thanks very much for your review and good suggestion.
“f string” is really cool a new feature from python 3.6.
Currently, we need to be compatible with python [3.5, 3.6, 3.7, 3.8] according to our [CI config](https://github.com/Project-MONAI/MONAI/blob/master/.github/workflows/release.yml#L15)
So can we mark down this and update it to "f string" in a future version?
Thanks.

Reply:
or shall we drop the py3.5 support? looks like we already have py3.6+ features in the codebase such as variable annotation